### PR TITLE
Adding new custom_connection_url parameter

### PR DESCRIPTION
**Pull request by maju6406**
custom_connection_url allows for edge case custom connection strings for more complex mssql installations. custom_connection_url takes precedence over all other connection related parameters.

### CODE REVIEW

**Comment by roobre** [comment]
Hi, thanks for your contribution!

Could you elaborate a bit more on the use cases where this is required?

I also wonder whether there could be an alternative approach that don't make one config flag to ignore others (as it is not very intuitive behavior), for example to add an `extra_connection_url_args` that could get concatenated to the generated URL.

Thanks!

**Comment by maju6406** [comment]
@roobre Good questions. The integration as-is doesn't allow for connecting to a read-only databases. To connect to a ready only database, you must set  'applicationintent=readonly'. The parameter was meant to be a manual override as the underlying library support much more parameters than this OHI exposes.

**Comment by roobre** [comment]
Thanks for elaborating the use case!
It seems like a corner case that wouldn't necessarily justify adding more flags. Allowing to specify custom parameters seems like an interesting feature which would be easy to implement. However, I'm still slightly concenred about the current implementation, as I think that overriding parameters is not very good UX.

Could you consider making this an "extra args" param? For example, the user could be instructed to supply url-encoded params (`applicationintent=readonly&foo=bar`), which could then be parsed in the code (for validation) and then added to the query.

**Comment by maju6406** [comment]
@roobre made the requested changes. Changed custom_connection_url to extra_connection_url_args

File: src/connection/sql_connection.go [hunk 1]
```diff
@@ -70,6 +73,16 @@ func CreateConnectionURL(args *args.ArgumentList) string {
 	query.Add("dial timeout", args.Timeout)
 	query.Add("connection timeout", args.Timeout)
 
+	if args.ExtraConnectionURLArgs != "" {
+		mockURL, err := url.Parse("http://example.com?" + args.ExtraConnectionURLArgs)
+		if err == nil {
```
Comment (gsanchezgavier) [inline src/connection/sql_connection.go:78]: why is this condition ? could you simplify the logic and use just url.ParseQuery (checking the output error if any)

File: src/connection/sql_connection.go [hunk 2]
```diff
@@ -70,6 +73,14 @@ func CreateConnectionURL(args *args.ArgumentList) string {
 	query.Add("dial timeout", args.Timeout)
 	query.Add("connection timeout", args.Timeout)
 
+	mockURL, err := url.Parse("http://example.com?" + args.ExtraConnectionURLArgs)
```
Comment (gsanchezgavier) [inline src/connection/sql_connection.go:76]: I mean the mockURL thing, I think we can remove it 
```suggestion
	if args.ExtraConnectionURLArgs != "" {
		extraArgsMap, err := url.ParseQuery(args.ExtraConnectionURLArgs)
		if err != nil {
			log.Warn("Unable to process ExtraConnectionURLArgs: %s", err.Error())
		}
		for k, v := range extraArgsMap {
			query.Add(k, v[0])
		}
	}
```

File: src/connection/sql_connection.go [hunk 3]
```diff
@@ -70,6 +73,13 @@ func CreateConnectionURL(args *args.ArgumentList) string {
 	query.Add("dial timeout", args.Timeout)
 	query.Add("connection timeout", args.Timeout)
 
+	if args.ExtraConnectionURLArgs != "" {
+		extraArgsMap, _ := url.ParseQuery(args.ExtraConnectionURLArgs)
```
[the remark this answers is not in the document]
Comment (maju6406) [reply src/connection/sql_connection.go]: Added a check and printing a warning if can't be parsed. I think it should still continue if this specific argument is wrong.

**Review by roobre — APPROVED** [review]

**Comment by roobre** [comment]
LGTM, thanks for working on this!

File: src/connection/sql_connection.go [hunk 4]
```diff
@@ -70,6 +73,18 @@ func CreateConnectionURL(args *args.ArgumentList) string {
 	query.Add("dial timeout", args.Timeout)
 	query.Add("connection timeout", args.Timeout)
 
+	if args.ExtraConnectionURLArgs != "" {
+		extraArgsMap, err := url.ParseQuery(args.ExtraConnectionURLArgs)
+		if err == nil {
```
Comment (gsanchezgavier) [inline src/connection/sql_connection.go:78]: just small sugestion for handling errors on Go, it common to check for != nil and handle the error below and if there is no error you can execute the code on the following lines.

**Review by gsanchezgavier — APPROVED** [review]

**Comment by gsanchezgavier** [comment]
Thank you very much for the contribution @maju6406  It has been merged !! 🥇

File: src/connection/sql_connection.go (hunk 4 above)
Comment (maju6406) [reply src/connection/sql_connection.go:78]: Thanks! This is my first golang contribution so I'm still learning. I appreciate the advice.

**Pull request merged**